tests-configuration-microprofile/src/test/java/org/jboss/hal/testsuite/test/configuration/microprofile/metrics/MicroprofileMetricsTest.java

```diff
@@ -13,32 +13,44 @@
 import org.jboss.dmr.ModelNode;
 import org.jboss.hal.testsuite.CrudOperations;
 import org.jboss.hal.testsuite.Random;
+import org.jboss.hal.testsuite.category.RequiresXP;
 import org.jboss.hal.testsuite.creaper.ManagementClientProvider;
 import org.jboss.hal.testsuite.dmr.ModelNodeGenerator;
 import org.jboss.hal.testsuite.page.configuration.MicroprofileMetricsPage;
+import org.junit.After;
 import org.junit.AfterClass;
 import org.junit.Test;
+import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 import org.openqa.selenium.WebDriver;
 import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
 import org.wildfly.extras.creaper.core.online.operations.Operations;
 
+import static org.jboss.hal.dmr.ModelDescriptionConstants.PREFIX;
 import static org.jboss.hal.testsuite.fixtures.microprofile.MicroprofileMetricsFixtures.EXPOSED_SUBSYSTEMS;
 import static org.jboss.hal.testsuite.fixtures.microprofile.MicroprofileMetricsFixtures.EXPOSE_ALL_SUBSYSTEMS;
 import static org.jboss.hal.testsuite.fixtures.microprofile.MicroprofileMetricsFixtures.MICROPROFILE_METRICS_ADDRESS;
 import static org.jboss.hal.testsuite.fixtures.microprofile.MicroprofileMetricsFixtures.SECURITY_ENABLED;
 
+@Category(RequiresXP.class)
 @RunWith(Arquillian.class)
 public class MicroprofileMetricsTest {
 
     private static final OnlineManagementClient client = ManagementClientProvider.createOnlineManagementClient();
     private static final Operations operations = new Operations(client);
+    private static final String STAR = "*";
 
     @AfterClass
     public static void closeClient() throws IOException {
+        operations.writeAttribute(MICROPROFILE_METRICS_ADDRESS, PREFIX, "false");
         client.close();
     }
 
+    @After
+    public void exposeAllSubsystems() throws Exception {
+        operations.writeListAttribute(MICROPROFILE_METRICS_ADDRESS, EXPOSED_SUBSYSTEMS, STAR);
```

Comment:
I agree with you. But in case of  `@After` the state is set to expose all subsystem which is indicated by `*` in this case `STAR`. In future when this value change the `STAR` value in the tests fail the test in the same way. But what can be done is clean up the in `@AfterClass` in case the future release change the exposed system from any substytem to specific subsytem.

Reply:
I had done it. Before class the the default exposed subsystems are stored and the exposed subsystems are set to `STAR`. As I mention before when condition change this create consistency for this test.